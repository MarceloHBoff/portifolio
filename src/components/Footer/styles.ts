import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Container = styled.div`
  ${({ theme }) => css`
    position: fixed;
    bottom: 0;
    left: 0;
    right: 0;
    height: 2.5rem;

    background: ${theme.colors.header};
    font-size: ${theme.font.sizes.small};

    padding: ${theme.spacings.xsmall} 30rem;

    ${media.lessThan('medium')`
      padding: ${theme.spacings.xsmall};
    `}
  `}
`
