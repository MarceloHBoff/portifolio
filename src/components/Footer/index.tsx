import { Container } from './styles'

const Footer = () => {
  return (
    <Container>
      <div>Â© Marcelo Boff 2021</div>
    </Container>
  )
}

export default Footer
