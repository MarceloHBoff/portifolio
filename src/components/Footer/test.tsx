import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils/tests/helpers'

import Footer from '.'

describe('<Footer />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Footer />)

    const footer = screen.getByText(/Â© Marcelo Boff 2021/i)
    expect(footer).toBeInTheDocument()
    expect(footer.parentElement).toHaveStyleRule('padding', '0.5rem', {
      media: '(max-width: 768px)'
    })

    expect(container.parentElement).toMatchSnapshot()
  })
})
