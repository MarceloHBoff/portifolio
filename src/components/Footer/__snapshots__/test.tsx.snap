// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
.c0 {
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  height: 2.5rem;
  background: #17202A;
  font-size: 1.4rem;
  padding: 0.5rem 30rem;
}

@media (max-width:768px) {
  .c0 {
    padding: 0.5rem;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <div>
        Â© Marcelo Boff 2021
      </div>
    </div>
  </div>
</body>
`;
